diff --git a/conf/battle/party.conf b/conf/battle/party.conf
index 863e0d2..d7bdb52 100644
--- a/conf/battle/party.conf
+++ b/conf/battle/party.conf
@@ -16,7 +16,7 @@ party_update_interval: 1000
 // Method used to update party-mate hp-bars:
 // 0: Aegis - bar is updated every time HP changes (bandwidth intensive)
 // 1: rAthena - bar is updated with the party map dots (up to 1 second delay)
-party_hp_mode: 0
+party_hp_mode: 1
 
 // Include SP values in party HP updates when supported by the client packet version. (Note 1)
 party_sp_on: yes
diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index 63bc244..45d5125 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -8250,7 +8250,7 @@ void clif_party_hp( const map_session_data& sd ){
 #else
 	p.hp = sd.battle_status.hp;
 	p.maxhp = sd.battle_status.max_hp;
-#if PACKETVER_ZERO_NUM >= 20210504
+#if PACKETVER_ZERO_NUM >= 20210504 || PACKETVER_MAIN_NUM >= 20210526 || PACKETVER_RE_NUM >= 20211103
 	p.sp = battle_config.party_sp_on ? sd.battle_status.sp : 0;
 	p.maxsp = battle_config.party_sp_on ? sd.battle_status.max_sp : 0;
 #endif
@@ -8308,7 +8308,7 @@ void clif_hpmeter_single( const map_session_data& sd, uint32 id, uint32 hp, uint
 #else
 	p.hp = hp;
 	p.maxhp = maxhp;
-#if PACKETVER_ZERO_NUM >= 20210504
+#if PACKETVER_ZERO_NUM >= 20210504 || PACKETVER_MAIN_NUM >= 20210526 || PACKETVER_RE_NUM >= 20211103
 	p.sp = battle_config.party_sp_on ? sp : 0;
 	p.maxsp = battle_config.party_sp_on ? maxsp : 0;
 #endif
