diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index 97ccf29..328bb43 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -10217,7 +10217,7 @@ void clif_name( const block_list* src, const block_list* bl, send_target target
 				safestrncpy( packet.position_name, md->guardian_data->castle->castle_name, NAME_LENGTH );
 
 				clif_send(&packet, sizeof(packet), src, target);
-			}else if( battle_config.show_mob_info ){
+			}else if( battle_config.show_mob_info || battle_config.mob_ele_view ){
 				PACKET_ZC_ACK_REQNAMEALL packet = { 0 };
 
 				packet.packet_id = HEADER_ZC_ACK_REQNAMEALL;
