diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index ef3a7dc..85b012f 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -2111,6 +2111,9 @@ void clif_move( const struct unit_data& ud )
 				clif_specialeffect(md, EF_GIANTBODY2, AREA);
 			else if (md->special_state.size == SZ_MEDIUM)
 				clif_specialeffect(md, EF_BABYBODY2, AREA);
+
+			if (battle_config.mob_ele_view)
+				clif_name_area(md);
 		}
 	break;
 	case BL_PET:
@@ -8253,10 +8256,6 @@ void clif_party_hp( const map_session_data& sd ){
 #else
 	p.hp = sd.battle_status.hp;
 	p.maxhp = sd.battle_status.max_hp;
-#if PACKETVER_ZERO_NUM >= 20210504 || PACKETVER_MAIN_NUM >= 20210526 || PACKETVER_RE_NUM >= 20211103
-	p.sp = battle_config.party_sp_on ? sd.battle_status.sp : 0;
-	p.maxsp = battle_config.party_sp_on ? sd.battle_status.max_sp : 0;
-#endif
 #endif
 
 	clif_send( &p, sizeof( p ), &sd, PARTY_AREA_WOS );
@@ -8311,10 +8310,6 @@ void clif_hpmeter_single( const map_session_data& sd, uint32 id, uint32 hp, uint
 #else
 	p.hp = hp;
 	p.maxhp = maxhp;
-#if PACKETVER_ZERO_NUM >= 20210504 || PACKETVER_MAIN_NUM >= 20210526 || PACKETVER_RE_NUM >= 20211103
-	p.sp = battle_config.party_sp_on ? sp : 0;
-	p.maxsp = battle_config.party_sp_on ? maxsp : 0;
-#endif
 #endif
 
 	clif_send( &p, sizeof( p ), &sd, SELF );
@@ -10191,11 +10186,11 @@ void clif_name( const block_list* src, const block_list* bl, send_target target
 					default: return "Unknown";
 				}
 			};
-			auto get_mob_size_name = [](int32 size) -> const char* {
+			auto get_mob_size_tag = [](int32 size) -> const char* {
 				switch (size) {
-					case SZ_SMALL: return "Small";
-					case SZ_MEDIUM: return "Medium";
-					case SZ_BIG: return "Large";
+					case SZ_SMALL: return "[S]";
+					case SZ_MEDIUM: return "[M]";
+					case SZ_BIG: return "[L]";
 					default: return "Unknown";
 				}
 			};
@@ -10217,17 +10212,17 @@ void clif_name( const block_list* src, const block_list* bl, send_target target
 				packet.packet_id = HEADER_ZC_ACK_REQNAMEALL_NPC;
 				packet.gid = bl->id;
 
-				char name_line[NAME_LENGTH] = {};
+				char title_line[NAME_LENGTH] = {};
 				if( battle_config.show_mob_info & 1 )
-					safesnprintf( name_line, sizeof(name_line), "%s [%u/%u]", md->name, md->status.hp, md->status.max_hp );
+					safesnprintf( title_line, sizeof(title_line), "%s [%u%%]", md->name, get_percentage( md->status.hp, md->status.max_hp ) );
 				else
-					safesnprintf( name_line, sizeof(name_line), "%s [%u%%]", md->name, get_percentage( md->status.hp, md->status.max_hp ) );
-				safestrncpy( packet.name, name_line, NAME_LENGTH );
-
-				char title_line[NAME_LENGTH] = {};
-				safesnprintf( title_line, sizeof(title_line), "%s Lv.%d | %s", get_mob_ele_name(md->status.def_ele), md->status.ele_lv, get_mob_size_name(md->status.size) );
+					safesnprintf( title_line, sizeof(title_line), "%s [%u%%]", md->name, get_percentage( md->status.hp, md->status.max_hp ) );
 				memcpy( packet.title, title_line, NAME_LENGTH );
 
+				char name_line[NAME_LENGTH] = {};
+				safesnprintf( name_line, sizeof(name_line), "%s %s", get_mob_ele_name(md->status.def_ele), get_mob_size_tag(md->status.size) );
+				safestrncpy( packet.name, name_line, NAME_LENGTH );
+
 				if (md->status.def_ele >= ELE_NEUTRAL && md->status.def_ele < ELE_MAX)
 					packet.groupId = 51 + md->status.def_ele;
 
@@ -10271,7 +10266,7 @@ void clif_name( const block_list* src, const block_list* bl, send_target target
 				if( battle_config.show_mob_info&2 )
 					str_p += sprintf( str_p, "HP: %u%% | ", get_percentage( md->status.hp, md->status.max_hp ) );
 				if( battle_config.mob_ele_view )
-					str_p += sprintf( str_p, "Ele: %s %d | Size: %s | ", get_mob_ele_name(md->status.def_ele), md->status.ele_lv, get_mob_size_name(md->status.size) );
+					str_p += sprintf( str_p, "Ele: %s | Size: %s | ", get_mob_ele_name(md->status.def_ele), get_mob_size_tag(md->status.size) );
 
 				if( str_p != mobhp ){
 					*(str_p-3) = '\0';
diff --git a/src/map/mob.cpp b/src/map/mob.cpp
index 58fd0e9..b2f7bfc 100644
--- a/src/map/mob.cpp
+++ b/src/map/mob.cpp
@@ -2788,7 +2788,7 @@ void mob_damage(mob_data *md, block_list *src, int32 damage)
 		mob_log_damage(md, src, static_cast<int64>(damage));
 	}
 
-	if (battle_config.show_mob_info&3)
+	if ((battle_config.show_mob_info&3) || battle_config.mob_ele_view)
 		clif_name_area(md);
 
 #if PACKETVER >= 20120404
@@ -3707,7 +3707,7 @@ void mob_revive(mob_data *md, uint32 hp)
 	clif_spawn(md);
 	skill_unit_move(md,tick,1);
 	mobskill_use(md, tick, MSC_SPAWN);
-	if (battle_config.show_mob_info&3)
+	if ((battle_config.show_mob_info&3) || battle_config.mob_ele_view)
 		clif_name_area(md);
 }
 
@@ -3904,7 +3904,7 @@ int32 mob_class_change (mob_data *md, int32 mob_id)
  *------------------------------------------*/
 void mob_heal(mob_data *md,uint32 heal)
 {
-	if (battle_config.show_mob_info&3)
+	if ((battle_config.show_mob_info&3) || battle_config.mob_ele_view)
 		clif_name_area(md);
 #if PACKETVER >= 20120404
 	if (battle_config.monster_hp_bars_info && !map_getmapflag(md->m, MF_HIDEMOBHPBAR)) {
